textEncodingTransformation.js

```diff
@@ -0,0 +1,41 @@
+const { readFileSync } = require('fs');
+
+const fileToTransform = 'node_modules/react-native-qrcode-svg/src/index.js';
+
+const upstreamTransformer = (() => {
+    try {
+      return require("@expo/metro-config/babel-transformer");
+    } catch (error) {
+      try {
+        return require("@react-native/metro-babel-transformer");
+      } catch (error) {
+        return require("metro-react-native-babel-transformer");
+      }
+    }
+  })();
+
+function createTransformer(transformer) {
+    return async ({src, filename, ...rest}) => {
+      if (filename === fileToTransform) {
+        const packageJsonPath = require.resolve('react-native/package.json');
+        const packageJson = JSON.parse(readFileSync(packageJsonPath, 'utf-8'));
+        const ReactNativeVersion = packageJson.version;
+        const [major, minor] = ReactNativeVersion.split('.');
+
+        // React Native versions below 0.75 do not include a global TextEncoder implementation.
+        // To ensure compatibility with these older versions, we add a polyfill using the 'text-encoding' library.
+        if (major === '0' && minor < '75') {
```

Comment:
what if React Native will be 1.0? 🙈 maybe we can leverage `semver` package here since we're in Node.js runtime
```suggestion
        if (semver.lt(version, '0.75.0')) {
```